butterfree/transform/features/feature.py

```diff
@@ -48,47 +52,65 @@ def __init__(
         self.dtype = dtype
         self.from_column = from_column
 
-    @property
-    def transformation(self) -> TransformComponent:
-        """Attribute transformation getter.
-
-        Returns
-            A transformation for this feature.
-        """
-        return self._transformation
-
-    @transformation.setter
-    def transformation(self, transformation: TransformComponent) -> None:
-        self._transformation = transformation
-        if transformation is not None:
-            self._transformation.parent = self
-
     @property
     def dtype(self) -> DataType:
-        """Attribute dtype getter.
-
-        Returns
-            The data type for this feature.
-        """
-        return self._dtype
+        """Attribute dtype getter."""
+        return self.__dtype
 
     @dtype.setter
-    def dtype(self, dtype: DataType) -> None:
+    def dtype(self, value: DataType) -> None:
         if (
             not (
                 isinstance(self.transformation, AggregatedTransform)
                 or isinstance(self.transformation, SparkFunctionTransform)
             )
-            and dtype is None
+            and value is None
         ):
             raise ValueError(
                 "dtype can't be None, except if the transformation is "
                 "AggregatedTransform or SparkFunctionTransform"
             )
-        if dtype is not None and not isinstance(dtype, DataType):
+
+        if value is not None and not isinstance(value, DataType):
             raise ValueError("dtype must be a DataType.")
 
-        self._dtype = dtype
+        self.__dtype = value
+
+    @property
+    def from_column(self) -> str:
+        """Attribute from_column getter."""
+        return self.__from_column
+
+    @from_column.setter
+    def from_column(self, value: str) -> None:
+        if (
+            isinstance(self.transformation, CustomTransform)
+            or isinstance(self.transformation, SQLExpressionTransform)
+            or isinstance(self.transformation, H3HashTransform)
+            or isinstance(self.transformation, StackTransform)
+        ) and value is not None:
+            raise ValueError(
```

Comment:
The code itself seems good to me. However, I would change this message to something like this: ```"from_column parameter doesn't need to be declared, since you've already declared it in our transformation. Besides, the column name {value} would be overwritten."``` What do you think?